app/controllers/api/v1/messages_controller.rb

```diff
@@ -139,12 +139,18 @@ def notification_serializer
       end
 
       def message_response(records, serializer)
-        ActiveModel::ArraySerializer.new(records,
+        ActiveModel::ArraySerializer.new(
+          records,
           each_serializer: serializer,
+          include_organisations_users: (staff? && params["include_organisations_users"] == "true"),
           root: "messages"
         ).as_json
       end
 
+      def staff?
+        current_user.has_permission?("can_manage_offers")
```

Comment:
What about stock app? Can we use `can_manage_offer_messages`, `can_manage_package_messages` and `can_manage_order_messages`

Reply:
This was specific to shared offer messages shown in admin app and code-change was introduced to load shared-offer message-thread to load charity-details. 
Hence admin-app users with permission `"can_manage_offers"` can see the chairty-details too of messgae-sender.